src/optimization/models/gridRelayOpt.cpp

```diff
@@ -34,115 +34,111 @@ gridRelayOpt::gridRelayOpt (const std::string &objName) : gridOptObject (objName
 
 gridRelayOpt::gridRelayOpt (coreObject *obj, const std::string &objName) : gridOptObject (objName),relay (dynamic_cast<Relay *> (obj))
 {
-  if (relay)
+    if (relay)
     {
-      if (getName().empty ())
+        if (getName().empty ())
         {
-          setName(relay->getName ());
+            setName(relay->getName ());
         }
-      setUserID(relay->getUserID ());
+        setUserID(relay->getUserID ());
     }
-
 }
 
 coreObject *gridRelayOpt::clone (coreObject *obj) const
 {
-  gridRelayOpt *nobj;
-  if (obj == nullptr)
+    gridRelayOpt *nobj;
+    if (obj == nullptr)
     {
-      nobj = new gridRelayOpt ();
+        nobj = new gridRelayOpt ();
     }
-  else
+    else
     {
-      nobj = dynamic_cast<gridRelayOpt *> (obj);
-      if (nobj == nullptr)
+        nobj = dynamic_cast<gridRelayOpt *> (obj);
+        if (nobj == nullptr)
         {
-          //if we can't cast the pointer clone at the next lower level
-          gridOptObject::clone (obj);
-          return obj;
+            //if we can't cast the pointer clone at the next lower level
+            gridOptObject::clone (obj);
+            return obj;
         }
     }
-  gridOptObject::clone (nobj);
-
-
-  //now clone all the loads and generators
-  //cloning the links from this component would be bad
-  //clone the generators and loads
-
+    gridOptObject::clone (nobj);
 
+    //now clone all the loads and generators
+    //cloning the links from this component would be bad
+    //clone the generators and loads
 
-  return nobj;
+    return nobj;
 }
 
 
 void gridRelayOpt::add (coreObject *obj)
 {
-  if (dynamic_cast<Relay *> (obj))
+    if (dynamic_cast<Relay *> (obj))
     {
-      relay = static_cast<Relay *> (obj);
-      if (getName().empty ())
+        relay = static_cast<Relay *> (obj);
+        if (getName().empty ())
         {
-          setName(relay->getName ());
+            setName(relay->getName ());
         }
-      setUserID(relay->getUserID ());
+        setUserID(relay->getUserID ());
+    }
+    else
+    {
+        throw(unrecognizedObjectException(this));
     }
-  else
-  {
-	  throw(unrecognizedObjectException(this));
-  }
 }
 
 count_t gridRelayOpt::objSize (const optimMode &)
 {
-  count_t objs = 0;
+    count_t objs = 0;
 
-  return objs;
+    return objs;
 }
 count_t gridRelayOpt::contObjSize (const optimMode &)
 {
-  count_t objs = 0;
+    count_t objs = 0;
 
-  return objs;
+    return objs;
 }
 
 count_t gridRelayOpt::intObjSize (const optimMode &)
 {
-  count_t objs = 0;
+    count_t objs = 0;
 
-  return objs;
+    return objs;
 }
 
 count_t gridRelayOpt::constraintSize (const optimMode &oMode)
 {
-  count_t objs = 0;
-  switch (oMode.linMode)
+    /*
```

Comment:
This looked unfinished too

Reply:
Most of the optimization library is unfinished